core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -105,11 +110,7 @@ public void run() {
           }, Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS),
           Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS), TimeUnit.MILLISECONDS);
 
-      try {
-        sessionHeartbeat();
-      } catch (InterruptedException e) {
-        throw Throwables.propagate(e);
-      }
+      NUM_ACTIVE_SESSIONS.incrementAndGet();
```

Comment:
the problem of keeping in the constructor is that it makes it unclear to the user whether the user should close the resource or not when the constructor throws exception. Without throwing exception in the ctor, it is very clear that the user of this class should always call close when an object is created.